src/core-services/product/resolvers/Query/products.js

```diff
@@ -0,0 +1,54 @@
+import getPaginatedResponse from "@reactioncommerce/api-utils/graphql/getPaginatedResponse.js";
+import wasFieldRequested from "@reactioncommerce/api-utils/graphql/wasFieldRequested.js";
+import { decodeProductOpaqueId, decodeShopOpaqueId, decodeTagOpaqueId } from "../../xforms/id.js";
+
+/**
+ * @name Query/products
+ * @method
+ * @memberof Products/Query
+ * @summary Query for a list of products
+ * @param {Object} _ - unused
+ * @param {Object} args - an object of all arguments that were sent by the client
+ * @param {String} args.shopId - id of user to query
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} info Info about the GraphQL request
+ * @returns {Promise<Object>} Products
+ */
+export default async function products(_, args, context, info) {
+  const {
+    productIds: opaqueProductIds,
+    shopIds: opaqueShopIds,
+    tagIds: opaqueTagIds,
+    query: queryString,
+    isArchived,
+    isVisible,
+    metafieldKey,
+    metafieldValue,
+    priceMin,
+    priceMax,
+    ...connectionArgs
+  } = args;
+
+  const shopIds = opaqueShopIds.map(decodeShopOpaqueId);
```

Comment:
I think the assumption here is that `shopIds` is required and must always have at least one shop ID. That would make sense because if you don't filter on shopId, they could view products for any shop, and the permission checks in the query function would be skipped. However, there is no `!` on `shopIds` field in the GraphQL schema, which means `opaqueShopIds` might be undefined/null here.

I think probably adding `!` is the best fix, as well as maybe a check in the query function to be sure there's at least one item in the array (or generally to ensure they can't access any product just by passing an empty array.

Reply:
Updated the schema and made the `shopIds` required